packages/main/src/RPA/Robocorp/WorkItems.py

```diff
@@ -948,50 +960,54 @@ def _load_adapter(self, default) -> Type[BaseAdapter]:
 
     @staticmethod
     def _interpret_content(body: str) -> Union[dict, str]:
-        loaders = [json.loads, yaml.full_load]
-        for loader in loaders:
+        loaders = {"JSON": json.loads, "YAML": yaml.full_load}
+        for name, loader in loaders.items():
```

Comment:
What about a dataclass implementation for the loaders?
This might grow in the future it would be easier to maintain the class than the current variant

Reply:
Can you provide a working example? As far as I remember, with dataclasses you usually create instances with the defined types which you can use as objects to map data inside. Here I just need a constant with which I want to iterate the loaders. ([stackoverflow](https://stackoverflow.com/a/59904697/4766178))

Quickest way without over-engineering would be to move this `loaders` outside of this function (as a `LOADERS` dict constant), but I want to keep it self-contained where's used only.

Reply with a working refactor and I'd be happy to integrate the improvement!